content/docs/iac/using-pulumi/mcp-server.md

```diff
@@ -0,0 +1,240 @@
+---
+title_tag: "Pulumi MCP Server | AI-Assisted Infrastructure as Code"
+meta_desc: "Learn how to use Pulumi's Model Context Protocol (MCP) server to integrate AI assistants like Cursor, Claude Code, and GitHub Copilot with your Pulumi workflow."
+title: MCP server
+h1: Pulumi Model Context Protocol server
+meta_image: /images/docs/meta-images/docs-meta.png
+weight: 6
+menu:
+    iac:
+        name: MCP server
+        parent: iac-using-pulumi
+        weight: 6
+---
+
+The Pulumi Model Context Protocol (MCP) server enables AI-powered coding assistants to seamlessly interact with Pulumi's CLI and registry, bringing real-time infrastructure knowledge and management directly into your development environment. This integration dramatically reduces context switching and accelerates Infrastructure as Code (IaC) development workflows.
+
+## What is the Pulumi MCP server?
+
+The [Model Context Protocol (MCP)](https://modelcontextprotocol.io) is a specification that allows language models to interact with external tools and data sources in a structured way. Pulumi's MCP server implementation provides AI assistants with the ability to:
+
+- **Query the Pulumi Registry** for resource information, properties, and documentation
+- **Execute Pulumi CLI commands** like `pulumi preview`, `pulumi up`, and `pulumi destroy`
+- **Access stack outputs** and configuration values
+- **Validate infrastructure code** before deployment
+- **Debug deployment issues** with real-time feedback
```

Comment:
We should also mention the fact that it can be used to catch and autofix IaC policy violations.